NPE on Azure Machine Creations

## Issue
@rfranzke observed NPEs while trying to create machines which didn't have an AzureMachineClass backing it. He mentioned that it might be related to quota issues? Need to further investigate.

## Logs

```
I0223 09:36:13.048282       1 machine_util.go:109] AzureMachineClass "shoot--vlabaz0006--ibgubkxfiq-worker-f7fie-26bde" not found. Skipping. azuremachineclass.machine.sapcloud.io "shoot--vlabaz0006--ibgubkxfiq-worker-f7fie-26bde" not found
I0223 09:36:13.098536       1 machine.go:315] Creating machine shoot--vlabaz0006--ibgubkxfiq-worker-comxa-5c646bdf8d-zssnx, please wait!
E0223 09:36:13.348035       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:522
/usr/local/go/src/runtime/panic.go:513
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/driver/driver_azure.go:80
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:317
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:184
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/machine.go:108
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:492
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:534
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:551
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/tmp/build/80754af9/git-gardener_machine-controller-manager-master_master/tmp/src/github.com/gardener/machine-controller-manager/pkg/controller/controller.go:513
/usr/local/go/src/runtime/asm_amd64.s:1333
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16a126a]
```

--- a/pkg/driver/driver_azure.go
+++ b/pkg/driver/driver_azure.go
@@ -77,7 +77,7 @@ func (d *AzureDriver) Create() (string, string, error) {
 		d.AzureMachineClass.Spec.SubnetInfo.SubnetName,
 		"",
 	)
-	err = onErrorFail(err, fmt.Sprintf("subnetClient.Get failed for subnet %q", *subnet.Name))
+	err = onErrorFail(err, fmt.Sprintf("subnetClient.Get failed for subnet %q", d.AzureMachineClass.Spec.SubnetInfo.SubnetName))
 	if err != nil {
 		return "Error", "Error", err
 	}

